[4966] Components options panel of page builder should open on the right side

aachavarria opened the pull request:
PreviewComponentsPanel.tsx: using pushPageBuilderPanelPage instead of pushToolsPanelPage, to open components tool pages;

adding target prop to createToolsPanelPage;

https://github.com/craftercms/craftercms/issues/4966

aachavarria committed:
PreviewComponentsPanel.tsx: using pushPageBuilderPanelPage instead of pushToolsPanelPage, to open components tool pages;
adding target prop to createToolsPanelPage;

rart reviewed (changes requested):
Please pull & merge.

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into bugfix/4966

aachavarria committed:
removing incorrect import

aachavarria committed:
using pushIcePanelPage instead of pushPageBuilderPanelPage

aachavarria committed:
adding EmptyState when there are not components

rart commented on ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx:
```diff
@@ -78,13 +78,7 @@ export default function PreviewComponentsPanel() {
-                defaultMessage="Communicate with your developers to create the required components in the system."
-              />
-            )
+            title: <FormattedMessage id="componentsPanel.emptyStateMessage" defaultMessage="No contentTypes found" />
```
- Why did you remove the subtitle?
- `contentTypes` is 2 words: "No content types found". We aren't listing content types here, are we? Think the original message was the right one. It's components that we're listing.

rart commented on ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx:
```diff
@@ -123,10 +117,12 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
+          createToolsPanelPage(
+            { id: 'previewBrowseComponentsPanel.title' },
+            [createWidgetDescriptor({ id: 'craftercms.components.PreviewBrowseComponentsPanel' })],
+            'pageBuilderPanel'
```
You sure you pulled? This has a new ID, it's icePanel. Check types after pulling merging. This shouldn't even compile.

rart replied:
Same for the other instances below (or on other files).

rart commented on ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx:
```diff
@@ -162,16 +162,28 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
+                defaultMessage="Communicate with your developers to create the required components in the system."
+              />
+            }
           />
```
Why is this in two different places

rart commented on ui/app/src/utils/state.ts:
```diff
@@ -122,11 +122,16 @@ export function getStoredGlobalAppOpenSidebar(user: string): string {
+export function createToolsPanelPage(
+  title: TranslationOrText,
+  widgets: WidgetDescriptor[],
+  target?: 'pageBuilderPanel' | 'toolsPanel'
```
```suggestion
  target?: 'icePanel' | 'toolsPanel'
```

rart reviewed (changes requested):
- See ToolsPanelPage.tsx:29, ToolsPanelPageButton.tsx:27, reducers/preview.ts:664. Please create type and type all of these instances with the same type.
- Call me if you'd like to review together.

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into bugfix/4966

aachavarria committed:
creating type ToolsPanelTarget;
renaming pageBuilderPanel target to icePanel;
fixing empty messages on PreviewComponentsPanel.tsx;

rart reviewed (changes requested):
Please adjust to what we discussed

aachavarria committed:
moving contentTypes filtering to top Parent component
